### ISSUE TITLE
operator-sdk init fails: only the go.mod and files with the prefix "(.)" are allowed before the init

### ISSUE THREAD

**Issue by bwoolf1** [operator-framework/operator-sdk#4618]
## Bug Report

The problem reported in issue #4355 is still a problem in operator-sdk v1.4.2. The issue has been closed but the problem hasn't been fixed.

#### What did you do?

```shell
~ operator-sdk init --domain=example.com --repo=github.com/example/memcached-operator
Error: failed to initialize project with "go.kubebuilder.io/v3": only the go.mod and files with the prefix "(.)" are allowed before the init
```

#### What did you expect to see?

Successfully initialize the project

#### What did you see instead? Under which circumstances?

Script fails because the directory is not completely empty. Directories like `.git` and `.idea` cause the script to fail.

#### Environment

**Operator type:**

/language go

**Kubernetes cluster type:**

OpenShift

`$ operator-sdk version`

operator-sdk version: "v1.4.2"

`$ go version` (if language is Go)

go version go1.15.8 darwin/amd64 (because Go v1.16 also doesn't work; see issue #4610)

`$ kubectl version`

Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"18d7461aca47e77cefb355339252a8d4c149188f", GitTreeState:"clean", BuildDate:"2021-01-30T16:44:37Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.3+e574db2", GitCommit:"e574db2", GitTreeState:"clean", BuildDate:"2021-01-30T06:33:00Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

#### Possible Solution

Ignore hidden directories like `.git` and `.idea`, the SDK should work anyway.

Also, don't close issues without fixing the problem and releasing a new version of the SDK with a fix.

#### Additional context

None.

**Comment by Adirio** [comment]
This has already been targeted in kubebuilder (the scaffolding library used underneath) to be a bit more permissive with which files are allowed. Directories and files starting with `.`, `go.mod`, `go.sum`, `README.md` and `LICENSE` [are allowed](https://github.com/kubernetes-sigs/kubebuilder/blob/2018a4f1c8dbcff9b868b80af03c7879c707edbe/pkg/plugins/golang/v3/init.go#L192-L230).
`.git` and `.idea` would fall under the directories starting by `.`.

Once SDK bumps the kubebuilder version, which is probably gated behing kubebuilder v3 release, these changes will be applied.

**Comment by jberkhahn** [comment]
This should be fixed by the latest version of Kubebuilder that will be included in 1.6.0 of Operator-sdk. Please reopen if not.
